pkg/operator/ceph/file/status.go

```diff
@@ -29,31 +29,41 @@ import (
 
 // updateStatus updates a fs CR with the given status
 func (r *ReconcileCephFilesystem) updateStatus(observedGeneration int64, namespacedName types.NamespacedName, status cephv1.ConditionType, info map[string]string) {
-	fs := &cephv1.CephFilesystem{}
-	err := r.client.Get(r.opManagerContext, namespacedName, fs)
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			logger.Debug("CephFilesystem resource not found. Ignoring since object must be deleted.")
-			return
+	const maxRetries = 3
+	var err error
+	retryMsg := " [retrying]"
+	for i := 0; i < maxRetries; i++ {
+		if i == maxRetries-1 {
+			// there will not be any retries
+			retryMsg = ""
+		}
+		fs := &cephv1.CephFilesystem{}
+		err = r.client.Get(r.opManagerContext, namespacedName, fs)
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				logger.Debug("CephFilesystem resource not found. Ignoring since object must be deleted.")
+				continue
+			}
+			logger.Warningf("failed to retrieve filesystem %q to update status to %q. %v %s", namespacedName, status, err, retryMsg)
+			continue
 		}
-		logger.Warningf("failed to retrieve filesystem %q to update status to %q. %v", namespacedName, status, err)
-		return
-	}
 
-	if fs.Status == nil {
-		fs.Status = &cephv1.CephFilesystemStatus{}
-	}
+		if fs.Status == nil {
+			fs.Status = &cephv1.CephFilesystemStatus{}
+		}
 
-	fs.Status.Phase = status
-	fs.Status.Info = info
-	if observedGeneration != k8sutil.ObservedGenerationNotAvailable {
-		fs.Status.ObservedGeneration = observedGeneration
-	}
-	if err := reporting.UpdateStatus(r.client, fs); err != nil {
-		logger.Warningf("failed to set filesystem %q status to %q. %v", fs.Name, status, err)
-		return
+		fs.Status.Phase = status
+		fs.Status.Info = info
+		if observedGeneration != k8sutil.ObservedGenerationNotAvailable {
+			fs.Status.ObservedGeneration = observedGeneration
+		}
+		if err = reporting.UpdateStatus(r.client, fs); err != nil {
+			logger.Warningf("failed to set filesystem %q status to %q. %v %s", fs.Name, status, err, retryMsg)
+			continue
+		}
```

Comment:
We are seeing that this sometimes fails, correct? Do we know why? What fields are changing between the get above (line 41) and here?

Reply:
FYI, you can use `kubectl get cephfilesystem --watch-only --output json` to have kubectl dump the json of an object over time as it changes. I usually run that before creating rook resources so I can see the initial and all changes as they happen. You have to manually diff the outputs though.

Reply:
Thanks @BlaineEXE for the `resource watch` command...